kolibri/core/assets/src/composables/__mocks__/useTotalProgress.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * `useTotalProgress` composable function mock.
+ *
+ * If default values are sufficient for tests,
+ * you only need call `jest.mock('<useTotalProgress file path>')`
+ * at the top of a test file.
+ *
+ * If you need to override some default values for some tests,
+ * or if you need to inspect the state of the refs during tests,
+ * you can import a helper function `useTotalProgressMock` that accepts
+ * an object with values to be overriden and use it together
+ * with  `mockImplementation` as follows:
+ *
+ * ```
+ * // eslint-disable-next-line import/named
+ * import useTotalProgress, { useTotalProgressMock } from '<useTotalProgress file path>';
+ *
+ * jest.mock('<useTotalProgress file path>')
+ * describe('describe test', function () {
+ *   let totalProgressMock = { totalProgress: ref(null) }
+ *
+ *   beforeAll(() => {
+ *     useTotalProgress.mockImplementation(() => useTotalProgressMock(totalProgressMock)
+ *   })
+ *
+ *   it('the test', () => {
+ *     expect(get(totalProgressMock.totalProgress)).toEqual(null);
+ *   )
+ * })
+ * ```
+ */
+import { ref, computed } from 'kolibri.lib.vueCompositionApi';
+import { get, set } from '@vueuse/core';
+import useUser from 'kolibri.coreVue.composables.useUser';
+import { UserProgressResource } from 'kolibri.resources';
+import { MaxPointsPerContent } from '../../constants';
+
+const MOCK_DEFAULTS = {
+  totalProgress: ref(null),
+};
+
+export function useTotalProgressMock(overrides = {}) {
+  const mocks = {
+    ...MOCK_DEFAULTS,
+    ...overrides,
+  };
+
+  const totalPoints = computed(() => mocks.totalProgress.value * MaxPointsPerContent);
+
+  const fetchPoints = () => {
+    const { isUserLoggedIn, currentUserId } = useUser();
+    if (get(isUserLoggedIn) && get(mocks.totalProgress) === null) {
+      UserProgressResource.fetchModel({ id: get(currentUserId) }).then(progress => {
```

Comment:
For the mock, we should be using a mock `jest.fn` callable instead of invoking the resource class itself, I think.

Reply:
Makes sense